### ISSUE TITLE
关于sparsebev_sampling中的sample_points_cam以及5-D tensor采样问题

### ISSUE THREAD

**Issue by kingofstu** [MCG-NJU/SparseBEV#74]
作者您好，我有两个疑问希望可以被解答
1、我看到sparsebev_sampling的代码中并未使用valid_mask，也就是说sample_points_cam中是存在投影不到相机平面的3d reference_points的，这样采样的时候不会有影响吗？padding会填充0对吗？
2、使用i_view[..., None].float() / (N - 1)拼接到sample_points_cam是因为each muvl_feats：[B*T*G, N, C, H, W]是5-D tensor，因此要cat相机索引到采样点中然后进行采样吗？那如果将each muvl_feats变为[B*T*G*N, C, H, W]是不是可以直接用4-D tensor了呢？

源码：
i_view = torch.argmax(valid_mask, dim=-1)[..., None]  # [B, T, Q, GP, 1]  对于每个3d点，只取其在一个相机的投影
sample_points_cam = sample_points_cam[i_batch, i_time, i_query, i_point, i_view, :]  # [B, Q, GP, 1, 2] 这里存在投射出图像平面外的3d点的投射坐标
valid_mask = valid_mask[i_batch, i_time, i_query, i_point, i_view]  # [B, Q, GP, 1]
sample_points_cam = torch.cat([sample_points_cam, i_view[..., None].float() / (N - 1)], dim=-1)

**Comment by WenxuanLi-whu** [comment]
你调试一下就知道了，无效的点的特征都是0
